src/queries/BaseQuery.ts

```diff
@@ -0,0 +1,45 @@
+import { PAGINATION_OFFSET_ROWS, PAGINATION_LIMIT } from '@/constants/constants';
+import { PaginationParams, getPaginationParams } from '@/lib/pagination';
+
+export interface QueryOptions {
+  isPaginated: boolean;
+}
+
+export interface QueryParams {
+  databaseInstance: any;
+  paginationParams?: PaginationParams;
+  options?: QueryOptions;
+}
+
+class BaseQuery {
+  databaseInstance: any;
+  paginationParams?: PaginationParams;
+  options: QueryOptions;
+  paginationQueryOptions: { [index: string]: any };
+
+  constructor(queryParams: QueryParams) {
+    this.databaseInstance = queryParams.databaseInstance;
+    this.paginationParams = queryParams.paginationParams;
+    this.options = queryParams.options || { isPaginated: true };
```

Comment:
If the only key in `queryParams.options` is `isPaginated`, maybe we can just:

```suggestion
    this.isPaginated = queryParams.isPaginated || true;
```